cpg_workflows/stages/joint_genotyping.py

```diff
@@ -67,17 +71,27 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
         vcf_path = self.expected_outputs(cohort)['vcf']
         siteonly_vcf_path = self.expected_outputs(cohort)['siteonly']
         scatter_count = joint_calling_scatter_count(len(cohort.get_sequencing_groups()))
+        # vcf framents, stripped of genotypes
         out_siteonly_vcf_part_paths = [
             to_path(
                 self.expected_outputs(cohort)['siteonly_part_pattern'].format(idx=idx)
             )
             for idx in range(scatter_count)
         ]
+        # vcf fragments, multiallelic variants split
+        out_split_vcf_part_paths = [
+            to_path(
+                self.expected_outputs(cohort)['split_part_pattern'].format(idx=idx)
+            )
+            for idx in range(scatter_count)
+        ]
 
         jc_jobs = joint_genotyping.make_joint_genotyping_jobs(
             b=get_batch(),
             out_vcf_path=vcf_path,
+            out_split_sitesonly_vcf_part_paths=out_split_vcf_part_paths,
```

Comment:
In the `make_joint_genotyping_jobs` changes in the `jobs/joint_genotyping`, two new arguments are added:
```
out_split_vcf_part_paths: list[Path] | None = None,
out_split_sitesonly_vcf_part_paths: list[Path] | None = None,
```

But only the latter is passed in here, does this check out? Is it because we want to trigger the `# if requested, make a site-only VCF for this split fragment (for VEP)` condition? And that we would only have a `out_split_vcf_part_paths` argument to pass in if the split vcf had already been made?

Reply:
Sorry, this is a bit convoluted. I confused myself a bit.

This process makes all the fragments separately and joins them up, with the code offering to save the end product, the fragments, or both. The files that we want at the end of this process are:

1. The whole VCF with split multiallelics and genotypes (for combining annotations -> Seqr)
2. The whole VCF without splitting, with no genotypes for VQSR
3. The fragments which are both split and no genotypes for VEP

All these VCF fragments are generated, presence in this list just determines which are generated in tmp, and which are persisted to GCP. I think it's correct to:

1. Not persist unsplit sites-only fragments (VQSR re-splits the whole sites-only VCF based on different logic. I think doing that is dumb, but the topic for another PR)
2. Persist split VCF with genotypes for annotateCohort later on
3. Persist the split & sites-only VCF fragments for feeding into VEP

Reply:
I've made a bunch of corrections to bring it in line with this intent ^ which it was definitely not doing before